File: hw/bonilla/U3/HW21TemplateMethod/TemplateMethodjs/model/Tea.js
const CaffeineBeverage = require('./CaffeineBeverage');
const readline = require('readline');

class Tea extends CaffeineBeverage {
  brew() {
    console.log("Steep the tea");
  }

  addCondiments() {
    console.log("Adding lemon");
  }

  wantsCondiments() {
    return new Promise(resolve => {
      const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
      });

      rl.question("Would you like lemon with your tea (y/n)? ", answer => {
        rl.close();
        resolve(answer.toLowerCase().startsWith('y'));
      });
    });
  }
}

module.exports = Tea;

File: hw/paillacho/U3/HW21TemplateMethod/TemplateMethod/python_version/template/model/tea.py
from caffeine_beverage import CaffeineBeverage
import sys
import os

# Author: Carlos Paillacho

sys.path.append(os.path.dirname(os.path.abspath(__file__)))


class Tea(CaffeineBeverage):

    def brew(self):
        print("Steeping the tea")

    def add_condiments(self):
        print("Adding lemon")

    def wants_condiments(self):
        answer = self.get_user_input()
        if answer.lower().startswith("y"):
            return True
        else:
            return False

    def get_user_input(self):
        try:
            answer = input("Would you like lemon with your tea (y/n)? ")
            return answer if answer else "n"
        except:
            return "n"
